lib/bundler/plugin.rb

```diff
@@ -121,13 +123,35 @@ def source_from_lock(locked_opts)
       src.new(locked_opts.merge("uri" => locked_opts["remote"]))
     end
 
+    def add_hook(event, &block)
+      (@hooks[event.to_s] ||= []) << block
```

Comment:
There where 3 places this hash was getting initialized, so I skipped this.

With calling `reset!` I think this can be used.